extension.js

```diff
@@ -16,10 +16,62 @@ const GREMLINS_SEVERITIES = {
   [GREMLINS_LEVELS.ERROR]: vscode.DiagnosticSeverity.Error,
 }
 
+const eventListeners = []
+
+let configuration = null
+
 let diagnosticCollection = null
 
-function gremlinsFromConfig(context) {
+function configureDiagnosticsCollection(showDiagnostics) {
+  if (showDiagnostics && !diagnosticCollection) {
+    diagnosticCollection = diagnosticCollection = vscode.languages.createDiagnosticCollection(GREMLINS)
```

Comment:
@TheSench what is the purpose of this?

Reply:
This function gets called during initialization and against if the user changes settings.  There is a single `DiagnosticCollection` created for the extension, and this function makes sure it exists when user wants to see `Diagnostics` (problems) and does not exist when they don't want to see them.

If the user unchecks the "Show Diagnostics" option, I want to remove all `Diagnostics` found and dispose of them.  If they re-check the box, I want to reinitialize the `DiagnosticCollection`.

If the user changed other Gremlins settings, I want to leave the `DiagnosticCollection` alone.

Reply:
ok